src/unistd.rs

```diff
@@ -2986,7 +2988,7 @@ impl From<User> for libc::passwd {
     }
 }
 
-#[cfg(not(target_os = "redox"))] // RedoxFS does not support passwd
+#[cfg(not(any(target_os = "redox", target_os = "haiku")))] // RedoxFS does not support passwd
```

Comment:
I see a definition for `getpwnam_r` in libc; can we make Haiku support all of these (including the `From` impls above)?

Reply:
sure, i gave it another shot and looks like this attempt was more successful than earlier attempts.